pkg/model/crd.go

```diff
@@ -359,52 +359,89 @@ func (r *CRD) ExceptionCode(httpStatusCode int) string {
-		out += fmt.Sprintf("%sreturn false", indent)
+//
+// if (r.ko.Spec.APIID == nil || r.ko.Status.RouteID != nil ) {
+//	  return true
+// } else {
+//	  return false
+// }
+func (r *CRD) GoCodeRequiredFieldsMissingFromShape(
+	opType OpType,
+	koVarName string,
+	indentLevel int,
+) string {
+	var op *awssdkmodel.Operation
+	switch opType {
+	case OpTypeGet:
+		op = r.Ops.ReadOne
+	case OpTypeGetAttributes:
+		op = r.Ops.GetAttributes
+	default:
+		return ""
 	}
-	return out
+
+	shape := op.InputRef.Shape
+	return r.goCodeRequiredFieldsMissingFromShape(
+		koVarName,
+		indentLevel,
+		shape,
+	)
 }
 
-// This method returns the required fields for ReadOneInput which are present in ko.Status .
-func (r *CRD) RequiredStatusFieldsForReadOneInput() []*CRDField {
-	var requiredStatusFields []*CRDField
-	op := r.Ops.ReadOne
-	inputShape := op.InputRef.Shape
-	if inputShape == nil || len(inputShape.Required) == 0 {
-		return requiredStatusFields
+func (r *CRD) goCodeRequiredFieldsMissingFromShape(
+	koVarName string,
+	indentLevel int,
+	shape *awssdkmodel.Shape,
+) string {
+	indent := strings.Repeat("\t", indentLevel)
+	if shape == nil || len(shape.Required) == 0 {
+		return fmt.Sprintf("%sreturn false", indent)
 	}
-	requiredFieldNames := inputShape.Required
-	for _, requiredFieldName := range requiredFieldNames {
-		koStatusField, found := r.StatusFields[requiredFieldName]
+	out := ""
+
+	// Loop over the required member fields in the shape and identity whether
+	// the field exists in either the Status or the Spec of the resource and
+	// generate an if condition checking for all required fields having non-nil
+	// corresponding resource Spec/Status values
+	missing := []string{}
+	for _, memberName := range shape.Required {
+		cleanMemberNames := names.New(memberName)
+		cleanMemberName := cleanMemberNames.Camel
+
+		resVarPath := koVarName
+		_, found := r.SpecFields[memberName]
 		if found {
-			requiredStatusFields = append(requiredStatusFields, koStatusField)
+			resVarPath = resVarPath + ".Spec." + cleanMemberName
+		} else {
+			_, found = r.StatusFields[memberName]
+			if !found {
+				// If it isn't in our spec/status fields, we have a problem!
+				msg := fmt.Sprintf(
+					"GENERATION FAILURE! there's a required field %s in "+
+						"Shape %s that isn't in either the CR's Spec or "+
+						"Status structs!",
+					memberName, shape.ShapeName,
+				)
+				panic(msg)
+			}
+			resVarPath = resVarPath + ".Status." + cleanMemberName
 		}
-	}
-	return requiredStatusFields
+		missing = append(missing, fmt.Sprintf("%s == nil", resVarPath))
+	}
+	// Use '||' because if any of the required fFields are missing the object
+	// is not created yet
+	missingCondition := strings.Join(missing, " || ")
+	out += fmt.Sprintf("%sif (%s) {\n", indent, missingCondition)
+	out += fmt.Sprintf("%s\treturn true\n", indent)
+	out += fmt.Sprintf("%s} else {\n", indent)
+	out += fmt.Sprintf("%s\treturn false\n", indent)
+	out += fmt.Sprintf("%s}", indent)
```

Comment:
nit: Maybe we can just return the `missingCondition` "value"
```go
out += fmt.Sprintf("%s\treturn %s\n", indent, missingCondition)
```

Reply:
ya, that's a good suggestion, @A-Hilaly.